src/solvers/flattening/boolbv_index.cpp

```diff
@@ -80,26 +80,16 @@ bvt boolbvt::convert_index(const index_exprt &expr)
     // this rather than as a series of individual options.
     #define UNIFORM_ARRAY_HACK
     #ifdef UNIFORM_ARRAY_HACK
-    bool is_uniform = false;
+    bool is_uniform = array.id() == ID_array_of;
 
-    if(array.id()==ID_array_of)
+    if(array.id() == ID_constant || array.id() == ID_array)
     {
-      is_uniform = true;
-    }
-    else if(array.id()==ID_constant || array.id()==ID_array)
-    {
-      bool found_exception = false;
-      forall_expr(it, array.operands())
-      {
-        if(*it != array.op0())
-        {
-          found_exception = true;
-          break;
-        }
-      }
-
-      if(!found_exception)
-        is_uniform = true;
+      is_uniform =
+        array.operands().size() <= 1 ||
+        std::all_of(
+          ++array.operands().begin(),
```

Comment:
:pick: `std::next` is a bit more readable IMO, also, for the sake of checking one extra element, perhaps just check them all